pkg/cloud/talos/provisioners/azure/azure.go

```diff
@@ -0,0 +1,415 @@
+	udConfigMap, err := utils.FetchConfigMap(cluster, machine, clientset)
+	if err != nil {
+		return err
+	}
+	ud := udConfigMap.Data["userdata"]
+	udb64 := base64.StdEncoding.EncodeToString([]byte(ud))
+
+	// Specify dummy val pass. We don't it anyways but it's required.
+	user := "talosuser"
+	password := utils.RandomString(15)
+
+	// Draft and create VM
+	vm := compute.VirtualMachine{
+		Location: to.StringPtr(azureConfig.Location),
+		VirtualMachineProperties: &compute.VirtualMachineProperties{
+			HardwareProfile: &compute.HardwareProfile{
+				VMSize: compute.VirtualMachineSizeTypes(azureConfig.Instances.Type),
+			},
+			StorageProfile: &compute.StorageProfile{
+				ImageReference: &compute.ImageReference{ID: to.StringPtr(azureConfig.Instances.Image)},
+				OsDisk: &compute.OSDisk{
+					OsType:       compute.Linux,
+					Name:         to.StringPtr(machine.ObjectMeta.Name + "-os-disk"),
+					CreateOption: compute.DiskCreateOptionTypesFromImage,
+					DiskSizeGB:   to.Int32Ptr(int32(azureConfig.Instances.Disks.Size)),
+				},
+			},
+			OsProfile: &compute.OSProfile{
+				ComputerName:  to.StringPtr(machine.ObjectMeta.Name),
+				AdminUsername: to.StringPtr(user),
+				AdminPassword: to.StringPtr(password),
+				CustomData:    to.StringPtr(udb64),
+			},
+			NetworkProfile: &compute.NetworkProfile{
+				NetworkInterfaces: &[]compute.NetworkInterfaceReference{
+					{
+						ID: nicObject.ID,
+						NetworkInterfaceReferenceProperties: &compute.NetworkInterfaceReferenceProperties{
+							Primary: to.BoolPtr(true),
+						},
+					},
+				},
+			},
+		},
+	}
+
+	vmClient, err := vmclient()
+	if err != nil {
+		return err
+	}
+
+	_, err = vmClient.CreateOrUpdate(ctx, azureConfig.ResourceGroup, machine.ObjectMeta.Name, vm)
+	if err != nil {
+		return err
+	}
+
+	log.Println("[Azure] Instance created: " + machine.ObjectMeta.Name)
+
+	return nil
+}
+
+//Update updates a given Azure instance.
+func (azure *Az) Update(ctx context.Context, cluster *clusterv1.Cluster, machine *clusterv1.Machine, clientset *kubernetes.Clientset) error {
+	return nil
+}
+
+// Delete deletes a Azure instance.
+func (azure *Az) Delete(ctx context.Context, cluster *clusterv1.Cluster, machine *clusterv1.Machine, clientset *kubernetes.Clientset) error {
+	machineSpec, err := utils.MachineProviderFromSpec(machine.Spec.ProviderSpec)
+	if err != nil {
+		return err
+	}
+
+	azureConfig := &ClusterInfo{}
+	yaml.Unmarshal([]byte(machineSpec.Platform.Config), azureConfig)
+
+	// Cleanup VM
+	vmClient, err := vmclient()
+	if err != nil {
+		return err
+	}
+	vmfuture, err := vmClient.Delete(ctx, azureConfig.ResourceGroup, machine.ObjectMeta.Name)
+	if err != nil {
+		return err
+	}
+	err = vmfuture.WaitForCompletionRef(ctx, vmClient.Client)
+	if err != nil {
+		return err
+	}
+
+	// Cleanup os disk
+	disksClient, err := disksclient()
+	if err != nil {
+		return err
+	}
+	disksfuture, err := disksClient.Delete(ctx, azureConfig.ResourceGroup, machine.ObjectMeta.Name+"-os-disk")
```

Comment:
Went ahead and waited for the completion of these guys with the future. Shouldn't take that long for these anyways, it's the VMs that really seem to take a while.